dear_petition/petition/models.py

```diff
@@ -324,3 +324,12 @@ def has_attachments(self):
 
 # Look-alike Petition object used to support JSON data-driven petitions
 DataPetition = namedtuple("DataPetition", ["form_type", "data_only"], defaults=[True])
+
+
+class GeneratedPetition(TimeStampedModel):
+    user = models.ForeignKey(
+        User, null=True, related_name="generated_petitions", on_delete=models.SET_NULL
+    )
+    form_type = models.CharField(choices=FORM_TYPES, max_length=255)
+    number_of_charges = models.IntegerField()
+    batch = models.ForeignKey(Batch, on_delete=models.DO_NOTHING)
```

Comment:
I think `models.DO_NOTHING` will end up raising an IntegrityError when `Batch` objects are deleted nightly:

```suggestion
    batch_id = models.PositiveIntegerField()
```